pkg/systemd/services/timer-dialog.jsx

```diff
@@ -74,18 +74,76 @@
     const [repeatPatterns, setRepeatPatterns] = useState([]);
     const [specificTime, setSpecificTime] = useState("00:00");
     const [isSpecificTimeOpen, setSpecificTimeOpen] = useState(false);
-    const [submitted, setSubmitted] = useState(false);
     const [commandNotFound, setCommandNotFound] = useState(false);
-    const validationFailed = {};
+    const [validationFailed, setValidationFailed] = useState({});
+    const [validForm, setValidForm] = useState(false);
 
-    if (!name.trim().length || !/^[a-zA-Z0-9:_.@-]+$/.test(name))
-        validationFailed.name = true;
-    if (!description.trim().length)
-        validationFailed.description = true;
-    if (!command.trim().length || commandNotFound)
-        validationFailed.command = true;
-    if (!/^[0-9]+$/.test(delayNumber))
-        validationFailed.delayNumber = true;
+    const validate = async (event) => {
+        if (!event)
+            return;
+
+        const { id, value } = event.target;
+        let notValid = true;
+        setDialogError("");
+
+        switch (id) {
+        case 'servicename': {
+            if (value.trim().length && /^[a-zA-Z0-9:_.@-]+$/.test(value))
+                notValid = false;
+            break;
+        }
+        case 'description': {
+            if (value.trim().length)
+                notValid = false;
+            break;
+        }
```

Comment:
These 5 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18148-20230110-092821-56aa70de-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)